lib/cc/analyzer/engine_output.rb

```diff
@@ -39,14 +50,24 @@ def valid_json?
         parsed_output.present?
       end
 
+      def valid_with_type?(type)
+        parsed_output &&
+          parsed_output["type"].present? &&
+          parsed_output["type"].downcase == type
+      end
+
       def parsed_output
         @parsed_output ||= JSON.parse(raw_output)
       rescue JSON::ParserError
         nil
       end
 
       def validator
-        IssueValidator.new(parsed_output)
+        if issue?
+          IssueValidator.new(parsed_output)
+        elsif measurement?
+          MeasurementValidator.new(parsed_output)
```

Comment:
Perhaps we can try this if/when there's a third type, but both `#to_json` and `#validator` have knowledge of all possible document types. It might be nice to have an intermediate class to call `#to_json` and `#validator` _on_ instead.

Reply:
I also feel like the current pattern is ugly, and definitely would want to do that extraction if we add a third type. I don't see it as absolutely necessary to do for this current change, and didn't do it since I didn't come into this with a very strong idea of what that new type would be called/look like. If you have a good idea there, I'm happy to do it now.

Reply:
> If you have a good idea there, I'm happy to do it now.

I don't! So let's hold off on that.